pkg/apis/duck/v1/identity_lifecycle.go

```diff
@@ -33,3 +34,13 @@ func (s *IdentityStatus) MarkWorkloadIdentityFailed(cs *apis.ConditionSet, reaso
 	// It will be counted for conditionReady only if it is failed.
 	cs.Manage(s).MarkFalse(apis.ConditionReady, "WorkloadIdentityFailed", messageFormat, messageA...)
 }
+
+func (s *IdentityStatus) MarkWorkloadIdentityUnknown(cs *apis.ConditionSet, reason, messageFormat string, messageA ...interface{}) {
+	cs.Manage(s).MarkUnknown(IdentityConfigured, reason, messageFormat, messageA...)
+	// ConditionType IdentityConfigured is not included in apis.NewLivingConditionSet{}, so it is not counted for conditionReady.
+	// Set ConditionReady to be Unknown if the initial status of ConditionReady is Ready.
+	// If the initial status of ConditionReady is not Ready, we keep it as it.
+	if c := cs.Manage(s).GetCondition(apis.ConditionReady); c != nil && c.Status == corev1.ConditionTrue {
```

Comment:
You can probably directly use `c.IsTrue()` instead of  `c != nil && c.Status == corev1.ConditionTrue`

Reply:
And 
```
func (s *IdentityStatus) MarkWorkloadIdentityNotConfigured(cs *apis.ConditionSet, reason, messageFormat string, messageA ...interface{}) {
	cs.Manage(s).MarkUnknown(IdentityConfigured, reason, messageFormat, messageA...)
}
``` 
can be removed, as it is duplicated with `MarkWorkloadIdentityUnknown`, and there is no other place use it except test.
By the way, could also you change `MarkWorkloadIdentityConfigured` to be `MarkWorkloadIdentityReady`, just make the style consistent.

Reply:
Sure.